erpnext/accounts/doctype/payment_entry/payment_entry.py

```diff
@@ -348,6 +349,12 @@ def set_status(self):
 		else:
 			self.status = 'Draft'
 
+	def check_reference_details(self):
+		if not self.reference_no:
```

Comment:
@bhattdevarsh  combine if statement like 
instead of getting warning  twice let it be to combine in one statement.
```suggestion
		if not (self.reference_no and self.reference_date):
                frappe.throw("Please enter 'Cheque/Reference No' and 'Cheque/Reference Date' before submitting it.")
```